### PR TITLE
Upgrade PyTorch dependencies to latest

### PR DESCRIPTION
**Pull request by joshdevins**
In preparation for an 8.0 release, this updates PyTorch NLP dependencies
to more recent and latest minor versions. Amongst other things, this
introduces a fix from transformers that is helpful for text embedding
tasks with certain DPR models.

See: https://github.com/huggingface/transformers/issues/13670

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,10 +59,9 @@
     "scikit-learn": ["scikit-learn>=0.22.1,<1"],
     "lightgbm": ["lightgbm>=2,<4"],
     "pytorch": [
-        "huggingface-hub>=0.0.17,<1",
-        "sentence-transformers>=2.0.0,<3",
+        "sentence-transformers>=2.1.0,<3",
         "torch>=1.9.0,<2",
-        "transformers[torch]>=4.11.0<5",
+        "transformers[torch]>=4.12.0<5",
```
Comment (sethmlarson) [inline setup.py:64]: We need a comma between `4.12.0` and `<5`

**Review by sethmlarson — CHANGES_REQUESTED** [review]
Looks good, one issue I saw just now in the `transformers[torch]` dependency:

File: setup.py (hunk 1 above)
Comment (joshdevins) [reply setup.py:64]: Funny, this passes linter locally 🤔

**Review by sethmlarson — APPROVED** [review]

**Comment by joshdevins** [comment]
@sethmlarson I think the CI failures are unrelated to this change.

**Comment by joshdevins** [comment]
@sethmlarson Will this make it to the 8.0 release or does it need a backport to the right branch(es)?

**Comment by sethmlarson** [comment]
@joshdevins No backport needed!

**Pull request merged**